src/utils/normalization.ts

```diff
@@ -107,3 +107,30 @@ export const normalize = (payload: OperationsRepository) => {
 
   return { apis, documentation } as OperationsRepository;
 };
+
+export const emptyObject = (object: any, preserveValueKeys: string[], ignoreNestedKeys: string[]) => {
+  for (const key in object) {
+    if (typeof object[key] === 'object' && !ignoreNestedKeys.includes(key)) {
+      emptyObject(object[key], preserveValueKeys, ignoreNestedKeys);
+    } else {
+      // eslint-disable-next-line functional/immutable-data
```

Comment:
anyway to get rid of this 😅

Reply:
:laughing: :+1: 
The abstraction here may be overkill, but I like your effort :) 

Some devs use lodash for this purpose, but I personally avoid lodash as far as possible, even though it is convenient. I avoid lodash because it's an additional dependency and it is less performant than native methods for this kind of data.

I would do something like this here:
```
export const emptyObject = (object: any, preserveValueKeys: string[], ignoreNestedKeys: string[]) =>
  Object.fromEntries(Object.entries(object).map(([key, value]) => {
    if (typeof value === 'object' && !ignoreNestedKeys.includes(key)) {
      return [key, emptyObject(object[key], preserveValueKeys, ignoreNestedKeys)];
    } else {
      return [key, preserveValueKeys.includes(key) ? object[key] : emptyReturn(object[key])];
    }
  }));
```
Here I'm using `Object.entries` to return an array of key/value pairs and then I use `map` to manipulate those values without mutating the original object. `Object.entries` turns the key/value pairs back into an Object. There may be a better way of doing this that's still native. It's on my todo list to research it more.

Reply:
So I tried this but it had issues with arrays, it was trasforming array to objects so something like this
```
    "servers": [
      {
        "url": "https://api.coingecko.com/api/v3"
      }
    ]
```

ends up like this

```
    "servers": 
{ 
   0: {
        "url": ""
       }
},
```